runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -5270,11 +5270,579 @@ static TR::Register *inlineFPTrg1Src3(TR::Node *node, TR::InstOpCode::Mnemonic o
    return targetRegister;
    }
 
+/**
+ * @brief Generates instruction sequence for inlining java/util/zip/CRC32C.updateBytes and updateDirectByteBuffer
+ *
+ * @param[in] node: node
+ * @param[in] cg: CodeGenerator
+ * @param[in] isRawAddress: true if this is the call to updateDirectByteBuffer
+ * @return the result register
+ */
+static TR::Register *inlineCRC32CUpdateBytes(TR::Node *node, TR::CodeGenerator *cg, bool isRawAddress)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Node *thirdChild = node->getThirdChild();
+   TR::Node *fourthChild = node->getChild(3);
+
+   const bool isConstEnd = fourthChild->getOpCode().isLoadConst();
+   const bool isConstOff = thirdChild->getOpCode().isLoadConst();
+
+   TR::Register *inputReg = cg->gprClobberEvaluate(firstChild);
+   TR::Register *arrayReg = cg->evaluate(secondChild);
+   TR_ARM64ScratchRegisterManager *srm = cg->generateScratchRegisterManager();
+
+   if (isConstEnd && isConstOff && !isRawAddress)
+      {
+      const uint32_t end = fourthChild->getInt();
+      const uint32_t off = thirdChild->getInt();
+      const uint32_t len = end - off;
+      static const char *pCRC32ConstLenLoopUnrollThreshold = feGetEnv("TR_AArch64CRC32ConstLenLoopUnrollThreshold");
+      static const uint32_t loopUnrollThreshold = (pCRC32ConstLenLoopUnrollThreshold != NULL) ? atoi(pCRC32ConstLenLoopUnrollThreshold) : 64;
```

Comment:
`loopUnrollThreshold` can be 0 when the input for `atoi()` is not recognized as a number.
I don't think there is a problem in unrolling the loop in such a case, but please double-check.

Reply:
Actually, it seems the small value would cause a problem. I added the upper and lower bounds check for the threshold value.